service/policy/db/namespaces.go

```diff
@@ -386,3 +398,212 @@ func (c PolicyDBClient) RemovePublicKeyFromNamespace(ctx context.Context, k *nam
 		KeyId:       k.GetKeyId(),
 	}, nil
 }
+
+// validateRootCertificate validates that the PEM string is a valid PEM-encoded root certificate
+func validateRootCertificate(pemStr string) error {
```

Comment:
I don't see anywhere we are validating the certificate signature to be valid.  We need to add a call to `cert.CheckSignatureFrom(cert)` to validate the self signed signature and validate the cert has not been changed.